packages/api/es/indexer.js

```diff
@@ -344,30 +387,39 @@ async function reingest(g) {
  * @returns {Promise} object with response from the three indexer
  */
 async function handlePayload(event) {
-  let payload;
-  const source = get(event, 'EventSource');
+  const executionModel = new Execution();
+  const granuleModel = new Granule();
+  const pdrModel = new Pdr();
 
-  if (source === 'aws:sns') {
-    payload = get(event, 'Sns.Message');
-    payload = JSON.parse(payload);
-  }
-  else {
-    payload = event;
-  }
+  const payload = event.EventSource === 'aws:sns'
+    ? JSON.parse(event.Sns.Message)
+    : event;
+
+  const executionRecord = await buildExecutionRecord(executionModel, payload);
+  const pdrRecord = buildPdrRecord(pdrModel, payload);
+  const granuleRecords = await buildGranuleRecords(granuleModel, payload);
 
-  let executionPromise;
-  const e = new Execution();
-  if (['failed', 'completed'].includes(payload.meta.status)) {
-    executionPromise = e.updateExecutionFromSns(payload);
+  const recordsCount = flatten([executionRecord, granuleRecords, pdrRecord]).length;
+
+  if (recordsCount > 10 || inTestMode()) {
```

Comment:
DynamoDB in Localstack does not support transactions, so don't use them for unit tests.

Reply:
The 10 here should be some kind of const

Reply:
Also a comment on the significance of 10

Reply:
Fixed.